pyproject.toml

```diff
@@ -1,3 +1,3 @@
 [build-system]
-requires = ["setuptools >= 66.1.0"]  # minimum setuptools version supporting Python 3.12
+requires = ["setuptools >= 66.1.0, <= 72.1.0"]  # lower bound to support controller Python versions, upper bound for latest version tested at release
```

Comment:
I wouldn't patch this metadata bit. Instead, we could maintain a constraint file with the exact pin that would be updated periodically. That file would then be also included into sdist, letting the downstreams read the version number if they need to.

Most people would install from wheels anyway, and the only ones who would build from source are downstreams. I don't think that metadata is the right place for injecting this limitation.